Tests/DatadogTests/DatadogPrivate/URLSessionSwizzlerTests.swift

```diff
@@ -0,0 +1,410 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import _Datadog_Private
+@testable import Datadog
+
+class URLSessionSwizzlerTests: XCTestCase {
```

Comment:
Some of the tests in this suite should be a `Swizzler` tests, and here we should be only testing the interesting aspects of `URLSession` swizzling.

What IMO should be moved away:
* `testThirdPartySwizzler()`
* `testOriginalClassGetsSwizzledBy3rdParty()`
* `testSwizzlingTheSwizzled()`
* `testSwizzleMultipleTimes()`

The first two do test if our swizzling plays well with any other swizzling done on the object. The latter couple checks if an object can be swizzled multiple times.

Testing those on a concrete implementations of swizzling (`URLSessionSwizzler`) would introduce too many unit tests for the same logic (I mean, we'd be repeating the same in `UIViewControllerSwizzler`, `UIControlSwizzler` etc.).

Reply:
First 2 cases test if swizzled `URLSession` forwards `super` calls properly; technically that tests if `super_xxx` methods in `TemplateURLSession` are implemented properly.
`Swizzler` cannot solve this problem, therefore this cannot be its responsibility.

Last 2 cases again should not be `Swizzler`s responsibility.

As far as I see, we have 2 main misunderstandings:

1. `Swizzler` is not a base class nor an abstract class; it's more of a wrapper over `ObjectiveC.runtime`, set of helper/utility methods for swizzling. Having a base class would require us to streamline every kind of swizzling in the same way whereas we may need different kinds of swizzling behavior for different domains, we don't know yet.
2. We need to repeat almost all the test cases for every `XXXSwizzler` implementation. There is no way to know if swizzling is well-implemented without unit tests as we don't have compile-time safety at all (`TemplateClass` approach is _just an attempt_ to improve readability and compile-time safety)